### PR TITLE
fix: short-term fix for issue #322

### PR DESCRIPTION
**Pull request by sigua-cs**
## Description
Max value for Host.Counter. Short-term fix for issue #322
 

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Checklist

- [x] Linter passes correctly
- [ ] Add tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by sigua-cs** [comment]
Does it make sense to remove `s.host.dec()`  
https://github.com/ContentSquare/chproxy/blob/5872a429dd50e8c0a37646e4147eb1fcff217a9e/proxy.go#L235
since the decrement will be called by `defer s.dec()` https://github.com/ContentSquare/chproxy/blob/5872a429dd50e8c0a37646e4147eb1fcff217a9e/proxy.go#L107
for the host that returns BadGateway?

File: proxy.go [hunk 1]
```diff
@@ -23,8 +23,10 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-// tmpDir temporary path to store ongoing queries results
-const tmpDir = "/tmp"
+const (
+	tmpDir             = "/tmp"                // tmpDir temporary path to store ongoing queries results
+	defaultHostCounter = uint32((1 << 32) - 1) // default value for Host.Counter; Short-term fix issue #322
```
Comment (mga-chka) [inline proxy.go:28]: be-careful, with such a big value you'll likely overflow. 

You should put 1 << 31

Comment (sigua-cs) [reply proxy.go:28]: I'll test 1<<31
However, when it's 1<<32 `s.host.inc()` will be called https://github.com/ContentSquare/chproxy/blob/5872a429dd50e8c0a37646e4147eb1fcff217a9e/proxy.go#L100
as result value will be 0

Comment (mga-chka) [reply proxy.go:28]: yes, and if 2 queries happens at the same time the value will be 2^32 + 1 (i.e overflow)

**Comment by sigua-cs** [comment]
After internal discussion closing this PR. I'll reopen another one with short-term fix

**Pull request closed without merging**